File: content/snippets/go/language_basics/hashing_and_checksums.go
package main

import (
	"crypto/sha256"
	"fmt"
	"hash/crc32"
)

func main() {
	text := "hello-world"
	bytes := []byte(text)

	sha256Sum := sha256.Sum256(bytes)
	crc32Value := crc32.ChecksumIEEE(bytes)

	fmt.Println("text:", text)
	fmt.Printf("sha256: %x\n", sha256Sum)
	fmt.Println("crc32:", crc32Value)
}

File: content/snippets/cpp/language_basics/hashing_and_checksums.cpp
#include <functional>
#include <iostream>
#include <numeric>
#include <string>

int main() {
    std::string text = "hello-world";

    std::size_t hash_value = std::hash<std::string>{}(text);
    int checksum = std::accumulate(text.begin(), text.end(), 0);

    std::cout << "text: " << text << "\n";
    std::cout << "hash: " << hash_value << "\n";
    std::cout << "checksum: " << checksum << "\n";
    return 0;
}
